### PR TITLE
Counselors portal all

### PR DESCRIPTION
**Pull request by Garo-A**

### CODE REVIEW

File: app/controllers/counselor/counselors_controller.rb [hunk 1]
```diff
@@ -1,13 +1,16 @@
 class Counselor::CounselorsController < Counselor::PortalController
 
   def show
-    @counselor = User.find(current_user.id)
+    @counselor = authorize_current_counsilor
+    @groups = authorize_current_counsilor.groups
   end
 
   def edit
+    @counselor = authorize_current_counsilor
```
Comment (TheTempas) [inline app/controllers/counselor/counselors_controller.rb]: Misspelling of counselor throughout this file.

File: app/controllers/counselor/portal_controller.rb [hunk 2]
```diff
@@ -1,4 +1,13 @@
 class Counselor::PortalController < ApplicationController
   before_filter :authorize_counselor
 
+  def authorize_current_counsilor
```
Comment (TheTempas) [inline app/controllers/counselor/portal_controller.rb]: Misspelling here as well.

File: app/controllers/counselor/portal_controller.rb [hunk 3]
```diff
@@ -1,4 +1,13 @@
...
   before_filter :authorize_counselor
 
+  def authorize_current_counselor
+    if params[:id] = Counselor.find_by(user_id: current_user.id).id
+      Counselor.find(params[:id])
+    else
+      nil
+      flash[:notice] = "Looks like something went wrong, try again!"
+  end
+end
```
Comment (DerangedMind) [inline app/controllers/counselor/portal_controller.rb]: Small indentation problem here. Please line it up! :-)

File: app/views/counselor/counselors/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<h1> Edit <%= @counselor.user.first_name %> <%= @counselor.user.last_name %> Information </h1>
+
+<%= form_tag("/c/profile", method: :put) do %>
```
Comment (DerangedMind) [inline app/views/counselor/counselors/edit.html.erb]: SOOOOO boss.

Comment (Garo-A) [inline app/views/counselor/counselors/edit.html.erb]: LOLLL

File: app/controllers/counselor/portal_controller.rb (hunk 3 above)
Comment (Garo-A) [inline app/controllers/counselor/portal_controller.rb]: Sure thing, hang on

**Pull request merged**